org.scala-ide.sdt.core.tests/src/org/scalaide/core/ui/completion/CompletionTests.scala

```diff
@@ -102,21 +132,46 @@ abstract class CompletionTests extends TextEditTests with CompilerSupport {
             s"${completions.size} completions found:\n\n$completionList")
       }
 
-      completion.fold(
-          throw new ComparisonFailure(
-              s"The completion '$completionToApply' does not exist.", completionToApply, completionList)) {
-        completion => completion.applyCompletionToDocument(doc, unit, caretOffset, enableOverwrite) foreach {
-          case (cursorPos, applyLinkedMode) =>
-            caretOffset =
-              if (!applyLinkedMode)
-                cursorPos
-              else
-                applyLinkedModel(doc, cursorPos, completion.linkedModeGroups)
+      if (completionToApply.nonEmpty) {
+        val completion = findCompletion(completionToApply)
+
+        completion.fold(
+            throw new ComparisonFailure(
+                s"The completion '$completionToApply' does not exist.", completionToApply, completionList)) {
+          completion => completion.applyCompletionToDocument(doc, unit, caretOffset, enableOverwrite) foreach {
+            case (cursorPos, applyLinkedMode) =>
+              caretOffset =
+                if (!applyLinkedMode)
+                  cursorPos
+                else
+                  applyLinkedModel(doc, cursorPos, completion.linkedModeGroups)
+          }
         }
+
       }
 
-      // unload given unit, otherwise the compiler will keep type-checking it together with the other tests
-      unit.scalaProject.presentationCompiler(_.discardCompilationUnit(unit))
+      // Clean up so that subsequent tests find an empty environment:
+      val unitsToDelete = (unit +: preloadedUnits)
+      val sourceFilesToDelete = unitsToDelete.map(_.lastSourceMap().sourceFile)
+
+      project.presentationCompiler(_.askFilesDeleted(sourceFilesToDelete.toList).get)
+      unitsToDelete.foreach(_.resource().delete(true, new NullProgressMonitor()))
```

Comment:
Deleting the compilation units like this results in exceptions messages written to the console during the tests. Although the tests still pass, I think that this indicates that I should do this differently. Any suggestions?